src/wrapper.rs

```diff
@@ -316,11 +291,9 @@ impl DxcCompiler {
         };
 
         let mut compile_error = 0u32;
-        unsafe {
-            result.get_status(&mut compile_error);
-        }
+        let status_hr = unsafe { result.get_status(&mut compile_error) };
 
-        if result_hr == 0 && compile_error == 0 {
+        if !result_hr.is_err() && !status_hr.is_err() && compile_error == 0 {
             Ok(DxcOperationResult::new(result))
         } else {
             Err((DxcOperationResult::new(result), result_hr))
```

Comment:
The functions with a `DxcOperationResult` are up for a rework next: they're tricky to get right and even https://asawicki.info/news_1719_two_shader_compilers_of_direct3d_12 from my notes doesn't explain very clearly what to do, but #29 and #30 are going to make it easier to implement said refactor at least.